legacy/usart.c

```
@@ -182,65 +217,109 @@ static uint8_t calXor(uint8_t *buf, uint32_t len) {
+
+  uint16_t read_pos = usart_dma_rx_read_pos;
+
+  if (write_pos == read_pos) {
+    return;
   }
-  p_buf += len - 1;
-  if (!usart_read_bytes(p_buf, 1, USART_TIMEOUT)) {
-    return false;
+
+  uint16_t available = 0;
+  if (write_pos > read_pos) {
```

Clang (14.0.6)

[warning] if with identical then and else branches

(bugprone-branch-clone)